lib/parsers/text_parser.js

```diff
@@ -85,8 +85,15 @@ function compile(fields, options, config) {
       db: field.schema,
       table: field.table,
       name: field.name,
-      string: function() {
-        return _this.packet.readLengthCodedString(field.encoding);
+      string: function(encoding = field.encoding) {
+        if (field.columnType === Types.JSON && encoding === field.encoding) {
```

Comment:
Shouldn't you also check that encoding is not already utf8? I have a case where `field.encoding` is, and passing utf8 explicitly still raises the warning.